heif_benchmark.py

```diff
@@ -0,0 +1,231 @@
+        } for compress_type in COMPRESS_TYPES
+    }
+
+    test_begin = time.time()
+    print("Start time:", test_begin)
+
+
+    for compress_type in COMPRESS_TYPES:
+        print(f"-----------------COMPRESS TYPE {compress_type}--------------------")
+        for compress_level in range(INITIAL_COMPRESS_LEVEL, MAX_COMPRESS_LEVEL + 1):
+            min_time_ns = float("inf")
+            max_time_ns = 0
+            total_time_ns = 0
+            min_size_B = float("inf")
+            max_size_B = 0
+            total_size_B = 0
+            min_compression_ratio = float("inf")
+            max_compression_ratio = 0
+            total_compression_ratio = 0
+            current_result = \
+                results[f"compress_type_{compress_type}"][f"compress_level_{compress_level}"]
+            for frame_index in range(FRAME_COUNT):
+                img = Image.open(pathlib.Path(INPUT_PATH, f"{frame_index}.png"))
+                buffer = io.BytesIO()
+
+                # Running encode
+                gc.disable()
+                start = time.time_ns()
+                img.save(
+                    buffer,
+                    format="HEIF",
+                    compress_level=compress_level,
+                    compress_type=compress_type,
+                )
+                end = time.time_ns()
+                gc.enable()
+
+                # Save singular test results
+                time_ns = end - start
+                size_B = buffer.getbuffer().nbytes
+                compression_ratio = 100 * size_B / os.path.getsize(
+                    pathlib.Path(INPUT_PATH, f"{frame_index}.png"),
+                )
+
+                min_time_ns, max_time_ns = update_min_max(min_time_ns, max_time_ns, time_ns)
+                min_size_B, max_size_B = update_min_max(min_size_B, max_size_B, size_B)
+                min_compression_ratio, max_compression_ratio = update_min_max(
+                    min_compression_ratio,
+                    max_compression_ratio,
+                    compression_ratio,
+                )
+
+                total_time_ns += time_ns
+                total_size_B += size_B
+                total_compression_ratio += compression_ratio
+                test_result = {
+                    "time_ns": time_ns,
+                    "size_B": size_B,
+                    "size_ratio_compressed_to_original_%": compression_ratio
+                }
+                current_result["frame_data"].append(test_result)
+
+                # Save one image (this one has 2 landing pads in it) for reference
+                if frame_index == FRAME_TO_SAVE:
+                    img.save(
+                        pathlib.Path(OUTPUT_PATH, f"ct{compress_type}_cl{compress_level}.png"),
+                        format="PNG",
+                        compress_level=compress_level,
+                        compress_type=compress_type,
+                    )
```

Comment:
Here, use the HEIF encoding so that human examinors can see result of HEIF encoding for visual comparison.